### PR TITLE
Backend link to Frontend

### PR DESCRIPTION
**Pull request by ronaldcohen1611**
Linking of backend API and database to frontend logic and components. Api calls are now being safeguarded by ArticleHandler and queried by ArticleDistobutor for suitable use in the frontend.

### CODE REVIEW

**Comment by chrism1001** [comment]
image
currently getting this error

File: web/src/components/SlidingPanel/SlidingPanel.tsx
Comment (chrism1001) [inline web/src/components/SlidingPanel/SlidingPanel.tsx:1]: image

**Review by chrism1001 — CHANGES_REQUESTED** [review]
Getting an error for categoriesArticlesMap being undefined in categorieslistitems.tsx and slidingpanel.tsx

**Comment by rodrigomv29** [comment]
I am taking a look right now

**Comment by rodrigomv29** [comment]
Did you guys run `yarn rw prisma migrate dev`

**Review by rodrigomv29 — COMMENTED** [review]
Works fine for me right now. Just remember that for every change to the schema we have to migrate

**Comment by rodrigomv29** [comment]
I saw the new changes now.... now we have a problem with dbAuth

**Comment by rodrigomv29** [comment]
I think we need to generate a new secret

**Comment by rodrigomv29** [comment]
I think the command is yarn rw g secret

**Comment by rodrigomv29** [comment]
`yarn rw g secret`

File: web/src/components/SlidingPanel/SlidingPanel.tsx
Comment (ronaldcohen1611) [reply web/src/components/SlidingPanel/SlidingPanel.tsx:1]: I was able to fix those bugs with optional chaining for all the pages.

**Comment by ronaldcohen1611** [comment]
I was able to go through and fix most of the page rendering bugs, verifying by simulating clone down onto new machine by using my desktop. It should be good to go now.
image

**Comment by rodrigomv29** [comment]
new-homepage

**Comment by rodrigomv29** [comment]
dark-mode-homepage

File: api/db/schema.prisma
Comment (Ballslamic) [inline api/db/schema.prisma:1]: Changed authentication requirements for mapping errors.

File: api/src/graphql/articles.sdl.ts
Comment (Ballslamic) [inline api/src/graphql/articles.sdl.ts:1]: Changed authentication requirements for mapping + loading errors.

File: api/src/graphql/categories.sdl.ts
Comment (Ballslamic) [inline api/src/graphql/categories.sdl.ts:1]: Changed authentication requirements for mapping errors.

**Comment by Ballslamic** [comment]
Went through the site for:
- Logged-in and logged-out, light/dark theme, all nav options + search work; as well as redirection to the actual article site.

**Review by chrism1001 — APPROVED** [review]
Thank you for fixing the errors. Works fine on my end.

**Pull request merged**